app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -681,16 +736,19 @@ class BrowserWebViewClient @Inject constructor(
         request: WebResourceRequest?,
         error: WebResourceError?,
     ) {
-        error?.let {
-            val parsedError = parseErrorResponse(it)
-            if (parsedError != OMITTED && request?.isForMainFrame == true) {
-                start = null
-                webViewClientListener?.onReceivedError(parsedError, request.url.toString())
-            }
+        error?.let { webResourceError ->
+            val parsedError = parseErrorResponse(webResourceError)
             if (request?.isForMainFrame == true) {
+                if (parsedError != OMITTED) {
+                    if (this.start != null) {
+                        decrementAndUntrackLoad("errored", request.url.toString())
```

Comment:
Have you verified we don't get onPageFinished as well in this case? Also, do we want to only call `webViewClientListener?.onReceivedError(parsedError, request.url.toString())` is mainframe? The change seems unrelated to the purpose of this PR